Update notebooks.md

hirokikanamaru opened the pull request:
課題１を提出します。よろしくお願いします。

hirokikanamaru committed:
Update notebooks.md

課題１を提出します。よろしくお願いします。

oyamad commented:
連続変数の関数の補間のグラフも描いてみましょう．[ここ](https://lectures.quantecon.org/jl/julia_libraries.html#interpolation)とか[ここ](https://github.com/QuantEcon/QuantEcon.lectures.code/blob/master/optgrowth/linapprox.jl)の．

hirokikanamaru commented:
連続変数の関数のところ追加しました。

oyamad commented:
まず関数を定義してから，そこから `my_lin_interp` で interpolation を作ります．いまのコードでは順番が逆になっています．
[ここ](https://lectures.quantecon.org/jl/julia_libraries.html#interpolation)のコードをそのまま (`LinInterp` を `my_lin_interp` に置き換えるだけ) 使ってみましょう．(他のゼミ生の notebook も見てみましょう．)

hirokikanamaru commented:
Using Plots
plotlyjs()
とすると下記のようなエラーが表示されてしまいました。

WARNING: Couldn't initialize plotlyjs.  (might need to install it?)

if !Plots.is_installed("PlotlyJS")
    Pkg.add("PlotlyJS")
end
if !Plots.is_installed("Rsvg")
    Pkg.add("Rsvg")
end
import Blink
Blink.AtomShell.install()

INFO: To do a standard install of plotlyjs, copy and run this:

 これに対して、Pkg.addをしようとすると、

julia> Pkg.add("PlotlyJS")
ERROR: unknown package Julia required by MyInterpolations
 in requirements(::Dict{String,Base.Pkg.Types.VersionSet}, ::Dict{String,Base.Pkg.Types.Fixed}, ::Dict{String,Dict{VersionNumber,Base.Pkg.Types.Available}}) at .\pkg\query.jl:11
 in resolve(::Dict{String,Base.Pkg.Types.VersionSet}, ::Dict{String,Dict{VersionNumber,Base.Pkg.Types.Available}}, ::Dict{String,Tuple{VersionNumber,Bool}}, ::Dict{String,Base.Pkg.Types.Fixed}, ::Dict{String,VersionNumber}, ::Set{String}) at .\pkg\entry.jl:477
 in edit(::Function, ::String, ::Base.Pkg.Types.VersionSet, ::Vararg{Base.Pkg.Types.VersionSet,N}) at .\pkg\entry.jl:30
 in (::Base.Pkg.Entry.##2#5{String,Base.Pkg.Types.VersionSet})() at .\task.jl:360
 in sync_end() at .\task.jl:311
 in macro expansion at .\task.jl:327 [inlined]
 in add(::String, ::Base.Pkg.Types.VersionSet) at .\pkg\entry.jl:51
 in (::Base.Pkg.Dir.##2#3{Array{Any,1},Base.Pkg.Entry.#add,Tuple{String}})() at .\pkg\dir.jl:31
 in cd(::Base.Pkg.Dir.##2#3{Array{Any,1},Base.Pkg.Entry.#add,Tuple{String}}, ::String) at .\file.jl:48
 in #cd#1(::Array{Any,1}, ::Function, ::Function, ::String, ::Vararg{Any,N}) at .\pkg\dir.jl:31
 in add(::String) at .\pkg\pkg.jl:100

となってしまったのですが、どこに問題があるのでしょうか？

oyamad commented:
ちょっとよくわからないので，とりあえず `Pkg.update()` をやって，出力をここに貼ってください．
(それらの行は3つの backticks で囲う．)

hirokikanamaru commented:
'''
julia> Pkg.update()
INFO: Updating METADATA...
INFO: Updating cache of Blosc...
INFO: Updating cache of BinDeps...
INFO: Updating cache of StatsBase...
INFO: Updating cache of DataFrames...
INFO: Updating cache of Distributions...
INFO: Updating cache of LightGraphs...
INFO: Updating cache of Compat...
INFO: Updating cache of JSON...
INFO: Updating cache of PlotThemes...
INFO: Updating cache of DataFrames...
INFO: Updating MyInterpolations master... 680e78ea → 2944db8a
INFO: Updating Plots master... ff3ba18b → e8da23c3
INFO: Computing changes...
ERROR: unknown package Julia required by MyInterpolations
 in requirements(::Dict{String,Base.Pkg.Types.VersionSet}, ::Dict{String,Base.Pkg.Types.Fixed}, ::Dict{String,Dict{VersionNumber,Base.Pkg.Types.Available}}) at .\pkg\query.jl:11
 in resolve(::Dict{String,Base.Pkg.Types.VersionSet}, ::Dict{String,Dict{VersionNumber,Base.Pkg.Types.Available}}, ::Dict{String,Tuple{VersionNumber,Bool}}, ::Dict{String,Base.Pkg.Types.Fixed}, ::Dict{String,VersionNumber}, ::Set{String}) at .\pkg\entry.jl:477
 in update(::String, ::Set{String}) at .\pkg\entry.jl:458
 in (::Base.Pkg.Dir.##2#3{Array{Any,1},Base.Pkg.Entry.#update,Tuple{String,Set{String}}})() at .\pkg\dir.jl:31
 in cd(::Base.Pkg.Dir.##2#3{Array{Any,1},Base.Pkg.Entry.#update,Tuple{String,Set{String}}}, ::String) at .\file.jl:48
 in #cd#1(::Array{Any,1}, ::Function, ::Function, ::String, ::Vararg{Any,N}) at .\pkg\dir.jl:31
 in update() at .\pkg\pkg.jl:210
'''

と表示されました。

oyamad commented:
なんですかね．
いったん `Pkg.rm("MyInterpolations")` で `MyInterpolations` を消して (2回やった方がよいかも)，それから `Pkg.update()` をやって，それでエラーが出なければ，もう一回 `Pkg.clone("https://github.com/hirokikanamaru/MyInterpolations.jl")` をやってください．エラーが出たらまたここに貼ってください．

hirokikanamaru commented:
MyInterpolationsを消し、Pkg.update()は普通に回りました。その後、Pkg.cloneも回りました。
一方、PlotlyJSをadd及びbuildしようとすると以下のようなエラーが出てしまいました。

'''
julia> Pkg.add("PlotlyJS")
INFO: Cloning cache of Blink from https://github.com/JunoLab/Blink.jl.git
INFO: Cloning cache of Codecs from https://github.com/dcjones/Codecs.jl.git
INFO: Cloning cache of DocStringExtensions from https://github.com/JuliaDocs/DocStringExtensions.jl.git
INFO: Cloning cache of Hiccup from https://github.com/JunoLab/Hiccup.jl.git
INFO: Cloning cache of HttpCommon from https://github.com/JuliaWeb/HttpCommon.jl.git
INFO: Cloning cache of HttpParser from https://github.com/JuliaWeb/HttpParser.jl.git
INFO: Cloning cache of HttpServer from https://github.com/JuliaWeb/HttpServer.jl.git
INFO: Cloning cache of Juno from https://github.com/JunoLab/Juno.jl.git
INFO: Cloning cache of Lazy from https://github.com/MikeInnes/Lazy.jl.git
INFO: Cloning cache of Media from https://github.com/JunoLab/Media.jl.git
INFO: Cloning cache of Mustache from https://github.com/jverzani/Mustache.jl.git
INFO: Cloning cache of Mux from https://github.com/JuliaWeb/Mux.jl.git
INFO: Cloning cache of PlotlyJS from https://github.com/sglyon/PlotlyJS.jl.git
INFO: Cloning cache of WebSockets from https://github.com/JuliaWeb/WebSockets.jl.git
INFO: Installing Blink v0.5.2
INFO: Installing Codecs v0.3.0
INFO: Installing DocStringExtensions v0.3.3
INFO: Installing Hiccup v0.1.1
INFO: Installing HttpCommon v0.2.7
INFO: Installing HttpParser v0.2.0
INFO: Installing HttpServer v0.2.0
INFO: Installing Juno v0.2.7
INFO: Installing Lazy v0.11.6
INFO: Installing Media v0.2.7
INFO: Installing Mustache v0.1.4
INFO: Installing Mux v0.2.3
INFO: Installing PlotlyJS v0.6.2
INFO: Installing WebSockets v0.2.3
INFO: Building HttpParser
INFO: Attempting to Create directory C:\Users\Hiroki\.julia\v0.5\HttpParser\deps\downloads
INFO: Downloading file https://s3.amazonaws.com/julialang/bin/winnt/extras/libhttp_parser_2_7_1.zip
INFO: Done downloading file https://s3.amazonaws.com/julialang/bin/winnt/extras/libhttp_parser_2_7_1.zip
INFO: Attempting to Create directory C:\Users\Hiroki\.julia\v0.5\HttpParser
INFO: Directory C:\Users\Hiroki\.julia\v0.5\HttpParser already created

7-Zip [64] 9.20  Copyright (c) 1999-2010 Igor Pavlov  2010-11-18

Processing archive: C:\Users\Hiroki\.julia\v0.5\HttpParser\deps\downloads\libhttp_parser_2_7_1.zip

Extracting  usr\lib
Extracting  usr\lib\libhttp_parser32.dll
Extracting  usr\lib\libhttp_parser64.dll

Everything is Ok

Folders: 1
Files: 2
Size:       263925
Compressed: 91886
INFO: Building MbedTLS
Using system libraries...
INFO: Building PlotlyJS
=========================================================================================================[ ERROR: PlotlyJS ]=========================================================================================================

LoadError: automatic download failed (error: 2148270092): https://api.plot.ly/v2/plot-schema?sha1
while loading C:\Users\Hiroki\.julia\v0.5\PlotlyJS\deps\build.jl, in expression starting on line 1

=====================================================================================================================================================================================================================================

==========================================================================================================[ BUILD ERRORS ]===========================================================================================================

WARNING: PlotlyJS had build errors.

 - packages with build errors remain installed in C:\Users\Hiroki\.julia\v0.5
 - build the package(s) and all dependencies with `Pkg.build("PlotlyJS")`
 - build a single package by running its `deps/build.jl` script

=====================================================================================================================================================================================================================================
INFO: Package database updated

julia> Pkg.build("PlotlyJS")
INFO: Building HttpParser
INFO: Building MbedTLS
Using system libraries...
INFO: Building PlotlyJS
=========================================================================================================[ ERROR: PlotlyJS ]=========================================================================================================

LoadError: automatic download failed (error: 2148270092): https://api.plot.ly/v2/plot-schema?sha1
while loading C:\Users\Hiroki\.julia\v0.5\PlotlyJS\deps\build.jl, in expression starting on line 1

=====================================================================================================================================================================================================================================

==========================================================================================================[ BUILD ERRORS ]===========================================================================================================

WARNING: PlotlyJS had build errors.

 - packages with build errors remain installed in C:\Users\Hiroki\.julia\v0.5
 - build the package(s) and all dependencies with `Pkg.build("PlotlyJS")`
 - build a single package by running its `deps/build.jl` script

=====================================================================================================================================================================================================================================
'''

oyamad commented:
エラーメッセージで表示されている URL にアクセスしてみましょう．"Maintenance in progress" だそうです．またあしたやってみてください．
(エラーメッセージ等をペーストするときは3つの backtick で囲ってください．"backtick" がわからなければ Google さんに聞いてください．)

hirokikanamaru commented:
わかりました。ありがとございます。

hirokikanamaru commented:
Plotsのエラーが解消されたため、連続関数の部分を修正しました。

oyamad committed:
Merge branch 'master' into patch-1

oyamad commented:
よくできました．
